File: app/models/commands/if_command.rb
class Commands::IfCommand < Command
  attr_accessor :condition
  attr_accessor :then
  attr_accessor :else

  def initialize(condition, if_true, if_false = nil)
    @condition = condition
    @then = if_true if if_true
    @else = if_false if if_false
  end

  def next=(cmd)
    @then.last.next = cmd if @then
    @else.last.next = cmd if @else
    super
  end

  def run(session)
    if session.eval(@condition)
      @then || super
    else
      @else || super
    end
  end
end

File: broker/src/commands/if_command.erl
-module(if_command).
-export([run/2]).
-include("session.hrl").

run(Args, #session{js_context = JS}) ->
  Condition = proplists:get_value(condition, Args),
  Then = proplists:get_value(then, Args),
  Else = proplists:get_value(else, Args),
  case mozjs:eval(JS, Condition) of
    true -> run_branch(Then);
    _    -> run_branch(Else)
  end.

run_branch(undefined) -> next;
run_branch(Branch) -> {goto, Branch}.